backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/scheduling/utils/VdsCpuUnitPinningHelper.java

```diff
@@ -129,64 +106,42 @@ private void previewPinOfPendingExclusiveCpus(List<VdsCpuUnit> cpuTopology, Map<
             return cpusToBeAllocated;
         }
 
-        int vcpus = vm.getNumOfCpus();
-        int cpusLeft = vcpus;
-        int hostCoresPerSocket = host.getCpuCores() / host.getCpuSockets();
-        int hostThreadsPerCore = host.getCpuThreads() / host.getCpuCores();
-
-        for (int socket : getOnlineSockets(cpuTopology)) {
-            int coresInSockets = getCoresInSocket(cpuTopology, socket).size();
-            if (hostCoresPerSocket * hostThreadsPerCore == coresInSockets &&
-                    coresInSockets >= vcpus &&
-                    getFreeCpusInSocket(cpuTopology, socket).size() < vcpus) {
-                // we should fit into one socket
-                continue;
-            }
-            cpusLeft = allocateCores(cpuTopology, hostThreadsPerCore, socket, cpusLeft,
-                    cpusToBeAllocated, true, vm.getId(), vm.getCpuPinningPolicy());
-            if (cpusLeft > 0) {
-                // iterate again on the cores, take whatever we can.
-                cpusLeft = allocateCores(cpuTopology, hostThreadsPerCore, socket, cpusLeft,
-                        cpusToBeAllocated, false, vm.getId(), vm.getCpuPinningPolicy());
-            }
-            if (cpusLeft == 0) {
-                return cpusToBeAllocated;
-            }
-        }
-        return cpusLeft == 0 ? cpusToBeAllocated : new ArrayList<>();
+        List<VdsCpuUnit> cpusToBeAllocated = allocateDedicatedCpus(cpuTopology, vm);
+        return cpusToBeAllocated.size() == vm.getNumOfCpus() ? cpusToBeAllocated : null;
```

Comment:
we probably need to add null-check somewhere now

Reply:
added to `collectForHost`.